### PR TITLE
Lizards now gain a slight speed boost when resting at a cost of losing both arm functionality

### PR DESCRIPTION
**Pull request by ynot01**
### Intent of your Pull Request

/https://public-media.si-cdn.com/filer/6e/30/6e301885-a2c6-4dbb-b7a2-d8758e95d847/lizard.jpg)

#### Changelog

:cl:  
tweak: Lizards now gain a slight speed boost when resting at a cost of losing both arm functionality
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species_types/lizardpeople.dm [hunk 1]
```diff
@@ -38,6 +38,19 @@
 
 	return randname
 
+/// ALL FOURS
+/datum/species/lizard/spec_rest(mob/living/carbon/H, is_resting)
```
Comment (jcatjmeow) [inline code/modules/mob/living/carbon/human/species_types/lizardpeople.dm]:
```suggestion
/datum/species/lizard/proc/spec_rest(mob/living/carbon/H, is_resting)
```

**Comment by Hopekz** [comment]
Makes sense but should be test merged once you get this to work to see how it feels in-game.

**Review by jcatjmeow — CHANGES_REQUESTED** [review]

**Comment by Mesalikepie** [comment]
Lizards zoom away from sec

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -451,9 +451,11 @@
 
 	if(!resting)
 		set_resting(TRUE, FALSE)
+		spec_rest(src, TRUE)
```
Comment (jcatjmeow) [inline code/modules/mob/living/living.dm:454]: Ok, here's your issue. 
You're calling here spec_rest. What this actually means is "mob/living/spec_rest" of which a proc does not exist.

The only spec rest that exists is in datum/species/lizard

so, you need to make spec rest defined at the mob level instead. Or, you can make a temporary object like so
`var/datum/species/lizard/L = new()`
and then call the proc like this
`L.spec_rest(src, true)`

Comment (jcatjmeow) [reply code/modules/mob/living/living.dm:454]: havent tested this but im like 80% sure itll work

**Review by jcatjmeow — CHANGES_REQUESTED** [review]

**Comment by Mqiib** [comment]
Can this be toggleable so I don't immediately drop everything when I'm trying to crawl under braindeads to fix something?

**Comment by maxion12345** [comment]
Lizards now unable to use hands.

Because no thumbs

**Comment by ynot01** [comment]
> Can this be toggleable so I don't immediately drop everything when I'm trying to crawl under braindeads to fix something?

can't exactly carry things when you're on all fours can you

**Comment by jcatjmeow** [comment]
To be honest this should just be an option for all species, to scramble away fast

**Comment by UselessTheremin** [comment]
this is really funny i want this now. as a trash scalie lizard main, i approve *hiss ing and skittering away from sec like the little goblin you are.

**Comment by Marmio64** [comment]
if you add this, nerf the heat resistance lizards get, so they have more drawbacks in relation to their upsides.

**Comment by Mqiib** [comment]
Does this allow them to ignore the slowdown from crawling or does it make their crawl very slightly faster? If the former then this would be maybe balanced considering the speed bonus is very small, and if the latter then this is arguably a downside.

**Comment by ynot01** [comment]
> Does this allow them to ignore the slowdown from crawling or does it make their crawl very slightly faster? If the former then this would be maybe balanced considering the speed bonus is very small, and if the latter then this is arguably a downside.

as the code currently (non-functionally) stands they only get a slight speed boost over crawlspeed but i believe everyone thinks it's the latter so ill look into doing that

**Comment by Mqiib** [comment]
Are you saying it's supposed to be slightly-faster-than-normal-crawl speed or slightly-faster-than-run speed? Because, again, if it's only slightly boosted crawl speed (ie slightly faster than slow) for the trade-off of not being able to use hands while resting then, assuming it's not toggleable, this is a bad thing.

**Comment by patpol4** [comment]
Hot

**Comment by Reed0506** [comment]
this is pretty stupid and i dont like it.

lizards are far more human than animal and i cannot see a reason this would make sense.

**Comment by Jutastre** [comment]
> 
> 
> Can this be toggleable so I don't immediately drop everything when I'm trying to crawl under braindeads to fix something?

What if you got a speed boost only if your hands are already empty? So you can still crawl at normal speed while holding items.

**Comment by ynot01** [comment]
lost motivation and probably wont merge

**Pull request closed without merging**